statsd/client_config.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2012-2016 Eli Janssen
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file.
+
+package statsd
+
+import (
+	"fmt"
+	"time"
+)
+
+type ClientConfig struct {
+	// addr is a string of the format "hostname:port", and must be somethign
+	// validly parsable by net.ResolveUDPAddr.
+	Address string
+
+	// prefix is the statsd client prefix. Can be "" if no prefix is desired.
+	Prefix string
+
+	// ResInterval is the interval over which the addr is re-resolved.
+	// Do note that this /does/ add overhead!
+	// If you need higher performance, leave unset (or set to 0),
+	// in which case the address will not be re-resolved.
+	ResInterval time.Duration
```

Comment:
Also add a note that if this is set, your client will leak memory if you don't explicitly close it since the goroutine created will never stop (this is unlike the other ways to make a Client)

Reply:
all clients should close, or they would leak (some just more than others).
I believe this was always the case.

Added a few notes to the godoc strings about it though.